### PR TITLE
Convert katamari async tests to BDD

### PR DESCRIPTION
**Pull request by techtangents**
Related Ticket: TINY-7469

Description of Changes:
* BDD conversions

Pre-checks:
* ~[ ] Changelog entry added~
* [X] Tests have been added (if applicable)
* [X] Branch prefixed with `feature/` for new features (if applicable)

Review:
* ~[ ] Milestone set~
* [ ] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/katamari/src/test/ts/atomic/api/async/FutureResultTest.ts [hunk 1]
```diff
@@ -7,114 +8,116 @@ import { FutureResult } from 'ephox/katamari/api/FutureResult';
...
-}))));
-
-promiseTest('FutureResult: fromFuture', () => fc.assert(fc.asyncProperty(fc.integer(), (i) => new Promise((resolve, reject) => {
-  FutureResult.fromFuture<number, unknown>(Future.pure(i)).get((ii) => {
-    eqAsync('eq', Result.value(i), ii, reject, tResult());
-    resolve();
-  });
-}))));
+describe('atomic.katamari.ap.async.FutureResultTest', () => {
+  it('FutureResult: nu', () => fc.assert(fc.asyncProperty(arbResult(fc.integer(), fc.integer()), (r) => new Promise((resolve, reject) => {
```
Comment (lnewson) [inline modules/katamari/src/test/ts/atomic/api/async/FutureResultTest.ts:17]: Clean up the `FutureResult:` prefix?

File: modules/katamari/src/test/ts/atomic/api/async/LazyValueTest.ts [hunk 2]
```diff
@@ -16,138 +17,140 @@ const lazyCounter = () => {
   });
 };
 
-promiseTest('LazyValue: get', () => new Promise((resolve, reject) => {
-  const lazy = lazyCounter();
+describe('atomic.katamari.api.async.LazyValueTest', () => {
+  it('LazyValue: get', () => new Promise<void>((resolve, reject) => {
```
Comment (lnewson) [inline modules/katamari/src/test/ts/atomic/api/async/LazyValueTest.ts:21]: `LazyValue:` prefixes here as well?

**Review by lnewson — APPROVED** [review]

File: modules/katamari/src/test/ts/atomic/api/async/FutureResultTest.ts (hunk 1 above)
Comment (techtangents) [reply modules/katamari/src/test/ts/atomic/api/async/FutureResultTest.ts]: Fixed

File: modules/katamari/src/test/ts/atomic/api/async/LazyValueTest.ts (hunk 2 above)
Comment (techtangents) [reply modules/katamari/src/test/ts/atomic/api/async/LazyValueTest.ts]: Fixed

**Review by metricjs — APPROVED** [review]

**Pull request merged**